chunk.go

```diff
@@ -285,8 +285,8 @@ func (lcd *largeChunkDecoder) nextByte() byte {
 func (lcd *largeChunkDecoder) ensureBytes(n int) {
 	if lcd.rem <= n {
 		rbuf := make([]byte, defaultChunkBufferSize)
-		copy(rbuf, lcd.rbuf[lcd.ptr:])
-		add := lcd.fillBuffer(rbuf[lcd.ptr:])
+		off := copy(rbuf, lcd.rbuf[lcd.ptr:lcd.ptr+lcd.rem])
```

Comment:
wait, isn't `lcd.ptr+lcd.rem == len(lcd.rbuf)`? If not, can you leave a comment here?

Reply:
It isn't guaranteed that `fillBuffer` will actually fill the buffer...perhaps I should change the name 😅  I'll add a comment here either way, though.

Generally generic `Reader` interfaces have `read` and `read_exact` variants. Because we're pulling bytes one-by-one, and those bytes are coming from a streaming decompressor, I figured that it's fine if we _could_ underfill, since in practice we likely will not.

From the golang docs.

> Read reads up to len(p) bytes into p. It returns the number of bytes read (0 <= n <= len(p)) and any error encountered. Even if Read returns n < len(p), it may use all of p as scratch space during the call. If some data is available but not len(p) bytes, Read conventionally returns what is available instead of waiting for more.